app/models/heads_of_term.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class HeadsOfTerm < ApplicationRecord
+  belongs_to :planning_application
+  has_many :reviews, as: :owner, dependent: :destroy, class_name: "Review"
+  has_many :terms, extend: TermsExtension, dependent: :destroy
+  has_many :validation_requests, through: :terms
+
+  accepts_nested_attributes_for :terms, allow_destroy: true
+  accepts_nested_attributes_for :reviews
+
+  after_update :maybe_create_review
+
+  def current_review
+    reviews.order(:created_at).last
+  end
+
+  def latest_validation_request
+    validation_requests.max_by(&:notified_at)
+  end
+
+  def latest_validation_requests
+    validation_requests.group_by(&:owner_id).map { |id, vr| vr.max_by(&:notified_at) }
+  end
+
+  def latest_active_validation_requests
+    latest_validation_requests.select { |vr| vr.state != "cancelled" }
+  end
+
+  def send_notification?
+    validation_requests.open.none? { |request| request.notified_at.present? } ||
+      (validation_requests.open.any? && (validation_requests.open.order(:created_at).last&.notified_at&.<= 1.business_day.ago))
+  end
+
+  private
+
+  def maybe_create_review
+    return if current_review.nil?
+    return unless current_review.status == "updated" && current_review.review_status == "to_be_reviewed"
+
+    create_review
+  end
+
+  def create_review
+    Review.create!(assessor: Current.user, owner_type: "HeadsOfTerm", owner_id: id, status: "complete")
```

Comment:
Is this better to create directly through the association, then might not need the owner_id etc?
i.e. would doing `reviews.create ....` work?